Source/gui/SNTMessageView.swift

```diff
@@ -149,8 +150,56 @@ public func OpenEventButton(customText: String? = nil, action: @escaping () -> V
   .help("⌘ Return")
 }
 
+public func AuthorizeViaTouchID(
+  reason: String,
+  replyBlock: @escaping (Bool) -> Void
+) -> (Bool, NSError?) {
+  let context = LAContext()
+  let semaphore = DispatchSemaphore(value: 0)
+  var addRule: Bool = false
+  var error: NSError?
+
+  context.evaluatePolicy(
+    .deviceOwnerAuthenticationWithBiometrics,
+    localizedReason: reason
+  ) {
+    success,
+    authenticationError in
+    {
+      addRule = success
+      error = authenticationError as NSError?
+      semaphore.signal()
+    }()
+  }
+
+  semaphore.wait()
```

Comment:
AuthorizeViaTouchID is being called from inside SwiftUI so it's running on the main thread - we should never block the main thread. Given that the callback is being passed into this function to call, why not remove the return value and leave it to run asynchronously?

Reply:
The main issue here is that if the replyBlock call doesn't complete before the window is closed then the rules may not apply. 

This has led to some funky behavior with standalone mode in the past.

Previously I was using a callback from the replyBlock to close the window to avoid this.

Reply:
The solution was to schedule the window close on the main thread from the callback using dispatch.